### DISCUSSION TITLE
custom linemode

### DISCUSSION THREAD

**Discussion by niekdomi** [sxyazi/yazi#1985] [1. Q&A] [answered]
### Describe the question

I'm trying to implement a custom *Linemode* in Yazi. Based on my understanding, I should be able to access functions from `linemode.lua`. Ideally, I would like a solution like this:

```lua
function Linemode:file_info()
	return ui.Line("%s   %s   %s", self:permissions(), self:size(), self:mtime())
end
```

Here's an example from the documentation:

```lua
-- ~/.config/yazi/init.lua
function Linemode:size_and_mtime()
	local time = math.floor(self._file.cha.modified or 0)
	if time == 0 then
		time = ""
	elseif os.date("%Y", time) == os.date("%Y") then
		time = os.date("%b %d %H:%M", time)
	else
		time = os.date("%b %d  %Y", time)
	end

	local size = self._file:size()
	return ui.Line(string.format("%s %s", size and ya.readable_size(size) or "-", time))
end
```

But instead of rewriting the logic, I would like to call functions like `permissions()`, `size()`, or `mtime()` directly, if possible. 

Another attempt I made was this:

```lua
function Linemode:file_info()
	-- Get time of last modification
	local time = math.floor(self._file.cha.modified or 0)
	if time == 0 then
		time = ""
	elseif os.date("%Y", time) == os.date("%Y") then
		time = os.date("%b %d %H:%M", time)
	else
		time = os.date("%b %d  %Y", time)
	end

	-- Get file size or number of files in a folder
	local size = self._file:size()
	if size then
		size = ya.readable_size(size)
	else
		local folder = cx.active:history(self._file.url)
		size = folder and tostring(#folder.files) or ""
	end

	-- Get file permissions
	local permissions = self._file.cha:perm() or ""

	return ui.Line(string.format("%s   %s   %s", permissions, size, time))
end
```

However, the permissions part doesn't seem to work properly.

### Additional Question:
Is it possible to display folder sizes directly? While sorting folders by size is supported, it seems this feature isn't available in `linemode.lua` yet. Can this functionality be implemented for custom Linemodes?

### Validations

- [X] I have searched the existing discussions/issues

**Comment by sxyazi** [comment, marked as answer]
With the [latest nightly version of Yazi](https://yazi-rs.github.io/docs/installation#official-binaries), you can achieve this by simply:

```lua
function Linemode:file_info()
  return string.format("%s   %s   %s", self:permissions(), self:size(), self:mtime())
end
```

![screenshot-002170](https://github.com/user-attachments/assets/99e968f0-bb9e-4277-acfd-8a3d180d4867)
